### DISCUSSION TITLE
JSON Web Key Secret URI?

### DISCUSSION THREAD

**Discussion by zvictor** [supabase/supabase#5911] [Questions] [answered]
Hello everyone!

I have been looking for an IdP that I can use to support [Faunadb's security model](https://docs.fauna.com/fauna/current/security/).
They offer [integration with external identity providers](https://fauna.com/blog/choosing-an-authentication-strategy-with-fauna) and native [IdP-agnostic ABAC](https://docs.fauna.com/fauna/current/security/external/).

There are plenty of material on [how to connect faunadb and auth0](https://fauna.com/blog/setting-up-sso-authentication-in-fauna-with-auth0) 🤢 but nothing on Supabase.

# Can I use Supabase Auth solutions as the IdP for fauna?

I tried figuring it out by myself, but after setting up Supabase I got stuck missing a `JSON Web Key Secret URI`.
I couldn't find any good results for my search on `Supabase+JWKs`, and
the standard `<project-name>.supabase.co/.well-known/jwks.json` returns `{"message":"no Route matched with those values"}`

**Can Supabase provide the right interface to be the IdP of Fauna or am I missing something?**
**Is there support for JWKs in Supabase?**

**Comment by w3b6x9** [comment, marked as answer]
@zvictor we currently only support symmetric key but we have plans to add support for asymmetric key in the future.

**Reply by zvictor** [reply]
Thanks for the reply! I'll stay tuned.

**Reply by dinvlad** [reply]
Is there any ETA on symmetric key signature support?

**Reply by w3b6x9** [reply]
@dinvlad asymmetric key signature support is currently not on the roadmap.

**Reply by raph** [reply]
is there a ticket opened for asymetric key support? its also blocking integration with cognito auth

**Comment by paddyfink** [comment]
@zvictor Did you end up finding a workaround for this?

**Reply by ZeroTimeDrift** [reply]
When is this coming out, I can't use supabase until this is implemented

**Reply by zvictor** [reply]
Nope, I haven't found.
I stopped looking for it after a while as I had to abandon the project I was working on because of other reasons 😔

**Comment by marcus-sa** [comment]
When is it coming?

**Comment by w3b6x9** [comment]
you can track the progress of this here: https://github.com/orgs/supabase/discussions/12759

**Comment by hf** [comment]
Hey everyone, Supabase Auth and the platform don't yet support asymmetric JWTs. This however is soon going to become possible, so watch out for announcements.

**Comment by Yogesh-Dubey-Ayesavi** [comment]
Any updates on this, I want jwks Uri to use in gcp api gateway jwt auth.

**Comment by alrightsure** [comment]
Any update on this? This is currently a blocking feature for me.

**Comment by frankdagostino** [comment]
still waiting here :)

**Comment by shray-narang** [comment]
Anytime soon would be nice😔
